client/my-sites/email/email-management/home/email-plan.jsx

```diff
@@ -95,9 +90,7 @@ function EmailPlan( { domain, hideHeaderCake = false, selectedSite, source } ) {
 	const currentRoute = useSelector( getCurrentRoute );
 
 	const canAddMailboxes =
-		( getGSuiteProductSlug( domain ) || getTitanProductSlug( domain ) ) &&
-		getGSuiteSubscriptionStatus( domain ) !== 'suspended' &&
-		canCurrentUserAddEmail( domain );
+		getGSuiteSubscriptionStatus( domain ) !== 'suspended' && canCurrentUserAddEmail( domain );
```

Comment:
`domain.canCurrentUserAddEmail`  could be true whether a subscription exists or not.
However, `( getGSuiteProductSlug( domain ) || getTitanProductSlug( domain ) )`  is only going to be `truthy` when a subscription exists.

Reply:
That's true. But my reasoning was that `EmailPlan` will only render if there's an active subscription, as per [this](https://github.com/Automattic/wp-calypso/blob/36a3fe40acd8f05fbaee444145406f6ae44ff80b/client/my-sites/email/email-management/email-home.tsx#L133). 

But in any case, this is outside of the scope of the PR and there are probably more things to refactor in here if we look at that assumption (assuming that it's even 100% correct, which it might not be). Reverting this change to make sure that we can focus on the relevant stuff!